packages/react-storage/src/components/StorageBrowser/views/LocationDetailView/LocationDetailView.tsx

```diff
@@ -91,6 +73,15 @@ export function LocationDetailView({
     onToggleSearchSubfolders,
   } = useLocationDetailView({ onNavigate: onNavigateProp, onExit });
 
+  const messageControlContent = hasError
+    ? getListItemsResultMessage({
+        items: pageItems,
+        errorMessage: message ?? DEFAULT_ERROR_MESSAGE,
+      })
+    : getListItemsResultMessage({
+        items: pageItems,
+      });
```

Comment:
Can drop `DEFAULT_ERROR_MESSAGE`, that should be returned from the callback. We should also provide `query` for deriving search specific messages:

```suggestion
  const messageControlContent = getListItemsResultMessage({
    items: pageItems,
    errorMessage: message,
    query: searchQuery,
  });
```

We may need to update `getListItemsResultMessage` to return `undefined` by default